File: kawazoe/q2029/tyama_codeiq2029.lua
#!/usr/bin/env lua
while true do
	n=io.read("*n")
	if not n then
		break
	end
	if n==0 then
		break
	end
	r=0
	i=2
	while 1 do
		x=n-math.floor(i*(i-1)/2)
		if x<i then
			break
		end
		if x%i==0 then
			r=r+math.floor(x/i)
		end
		i=i+1
	end
	print(r)
end

File: kawazoe/q2029/tyama_codeiq2029.nim
#!/usr/bin/env nim
import strutils
while true:
 var s=readLine(stdin)
 if s=="":
  break
 #if not readLine(stdin,s):
 # break
 var n=cast[int64](parseBiggestInt(s))
 if n==0:
  break
 var r=cast[int64](0)
 var i=cast[int64](2)
 while true:
  var x=n-(i*(i-1) div 2)
  if x<i:
   break
  if x mod i==0:
   r=r+(x div i)
  i=i+1
 echo(r)